### DISCUSSION TITLE
Rotated texts

### DISCUSSION THREAD

**Discussion by Andreasox** [pymupdf/PyMuPDF#3011] [Looking for help] [answered]
Is it possible to create a text with any rotation angle? insert_text only accepts values 90, 180,270. We need to rotate texts 45 degrees.

**Comment by JorjMcKie** [comment, marked as answer]
Method [insert_textbox](https://pymupdf.readthedocs.io/en/latest/page.html#Page.insert_textbox) supports the `morph=(point, matrix)` parameter.
If you do `page.insert_textbox(rect, text, morph=(rect.bl, fitz.Matrix(45)), ...)`, the text will be rotated by 45° around the bottom-left point of the rectangle.

**Reply by JorjMcKie** [reply]
`page.insert_text()` also has that `morph` parameter.
Please not that the point in `morph=(point, matrix)` can be anywhere on the page, and the matrix need not be a rotation but could be any matrix with `matrix.e = matrix.f = 0`. In this way, one can achieve up-down / left / right text flipping and much more.
The `point` serves as a fix-point here.

**Comment by JorjMcKie** [comment]
Other interesting effects are selective stretching / shrinking of the text. For example, using the matrix `Matrix(2, 0.5)` stretches by factor 2 in horizontal, and by factor 0.5 in vertical direction.
Look at these [eamples](https://pymupdf.readthedocs.io/en/latest/matrix.html#examples).

**Reply by Andreasox** [reply]
Thanks a lot. This works for our use case. 
I assume that matrix do not respect kerning rules, i.e. stretch is purely math calculation.

**Reply by JorjMcKie** [reply]
Yes, of course.
